### DISCUSSION TITLE
SetValue to Array

### DISCUSSION THREAD

**Discussion by skeathly** [react-hook-form/react-hook-form#2987] [Q&A]
Looking to use setValue to update to array of objects like so:

setValue('thing', [{thing: 'thing'}])

but getting back
{
    thing: "[object, Object]"
}

Wondering what I can do to fix this issue?

Thanks.

**Comment by bluebill1049** [comment]
you have a codesandbox for it?

**Comment by skeathly** [comment]
I put this simple example together but realized this isn't an issue with setValue, but an issue of trying to set an input to object value.  Sorry I believe this is more of an issue of getting started using React.  If you still have a solution of how to handle these cases that would be much appreciated.

https://codesandbox.io/s/hook-form-full-name-from-object-tn53x?file=/src/index.js

**Reply by skeathly** [reply]
![image](https://user-images.githubusercontent.com/4078663/95399120-0fb88180-08cd-11eb-9020-7957a6e395aa.png)
 I've attached an image to illustrate what I'm trying to do - sounds like React Hook Form does not allow to work like this.

**Reply by bluebill1049** [reply]
setValue is for a single input value, so for input you will target the name and set an input with a string of value. this would apply the same to your checkbox, is your checkboxes have the same name?
